ntbs-service/Pages/Notifications/Edit/HospitalDetails.cshtml.cs

```diff
@@ -175,8 +187,29 @@ private async Task GetRelatedEntities()
 
         public async Task<JsonResult> OnGetGetFilteredListsByTbService(string value)
         {
-            var filteredHospitalDetailsPageSelectLists =
-                await _referenceDataRepository.GetFilteredHospitalDetailsPageSelectListsByTbService(value);
+            var notification = await NotificationRepository.GetNotificationAsync(NotificationId);
+            var tbServiceCodeAsList = new List<string> { value };
+
+            var filteredHospitals =
+                await GetActiveOrCurrentHospitalsByTbServiceCodesAsync(tbServiceCodeAsList, notification);
+
+            var filteredCaseManagers =
+                await _referenceDataRepository.GetCaseManagersByTbServiceCodesAsync(tbServiceCodeAsList);
+
+            var filteredHospitalDetailsPageSelectLists = new FilteredHospitalDetailsPageSelectLists
+            {
+                Hospitals = filteredHospitals.Select(n => new OptionValue
+                {
+                    Value = n.HospitalId.ToString(),
+                    Text = n.Name
+                }),
+                CaseManagers = filteredCaseManagers.Select(n => new OptionValue
+                {
+                    Value = n.Id.ToString(),
+                    Text = n.DisplayName
+                })
+            };
```

Comment:
Another sign that this is a good refactor is that mapping like this is much more suited to the view model (where we're preparing data for being viewed on the page) than in the repository (where we're meant to just be fetching data from the database).